### PR TITLE
Remove message about `pub cache` that is not actionable

### LINKED ISSUE #140628: `flutter create` emits confusing language about caches unprompted
Version is `3.16.5` (stable):

```shell
$ flutter --version
Flutter 3.16.5 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 78666c8dc5 (7 days ago) • 2023-12-19 16:14:14 -0800
Engine • revision 3f3e560236
Tools • Dart 3.2.3 • DevTools 2.28.4
```

## Example

```shell
$ flutter create --empty --platforms macos my_app
Found an existing Pub cache at /Users/matan/.pub-cache.
It can be repaired by running `dart pub cache repair`.
It can be reset by running `dart pub cache clean`.
Found an existing Dart Analysis Server cache at /Users/matan/.dartServer.
It can be reset by deleting /Users/matan/.dartServer.
```

---

I have no idea what I'm supposed to be doing with this information, or why it was presented to me. Are my caches stale? What would lead me to wanting to repair or reset either of these caches?

### PR DESCRIPTION
**Pull request by matanlurey**
... and neither the pub nor tools team think it's important.

Fixes https://github.com/flutter/flutter/issues/140628.

### CODE REVIEW

File: analysis_options.yaml [hunk 1]
```diff
@@ -20,8 +20,10 @@ analyzer:
     strict-inference: true
     strict-raw-types: true
   errors:
-    # allow self-reference to deprecated members (we do this because otherwise we have
-    # to annotate every member in every test, assert, etc, when we deprecate something)
+    # allow deprecated members (we do this because otherwise we have to annotate
```
Comment (jonahwilliams) [inline analysis_options.yaml:23]: Nit: unrelated?

Comment (matanlurey) [reply analysis_options.yaml:23]: Oh oops, thanks!

**Review by christopherfujino — APPROVED** [review]
LGTM
